specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/costmanagement.generatedetailedcostreport.json

```diff
@@ -0,0 +1,373 @@
+      "type": "object",
+      "properties": {
+        "error": {
+          "description": "The details of the error.",
+          "$ref": "common-types.json#/definitions/ErrorDetails"
+        }
+      }
+    },
+    "GenerateDetailedCostReportDefinition": {
+      "description": "The definition of a cost detailed report.",
+      "properties": {
+        "metric": {
+          "description": "The type of the detailed report. By default ActualCost is provided",
+          "type": "string",
+          "enum": [
+            "ActualCost",
+            "AmortizedCost"
+          ],
+          "x-ms-enum": {
+            "name": "GenerateDetailedCostReportMetricType",
+            "modelAsString": true
+          }
+        },
+        "timePeriod": {
+          "description": "Has time period for pulling data for the cost detailed report. Can only have one of either timePeriod or invoiceId or billingPeriod parameters. If none provided current month cost is provided.",
+          "$ref": "#/definitions/GenerateDetailedCostReportTimePeriod"
+        },
+        "billingPeriod": {
+          "description": "Billing Period in YearMonth(e.g. 202008) format. Only for legacy enterprise customers can use this. Can only have one of either timePeriod or invoiceId or billingPeriod parameters. If none provided current month cost is provided.",
+          "type": "string"
+        },
+        "invoiceId": {
+          "description": "Invoice Id for PayAsYouGo customers and Modern billing profile scope. Can only have one of either timePeriod or invoiceId or billingPeriod parameters. If none provided current month cost is provided.",
+          "type": "string"
+        },
+        "customerId": {
+          "description": "Customer Id for Modern (Invoice Id and billing profile is also required for this).",
+          "type": "string"
+        }
+      }
+    },
+    "GenerateDetailedCostReportTimePeriod": {
+      "description": "The start and end date for pulling data for the cost detailed report.",
+      "properties": {
+        "start": {
+          "description": "The start date to pull data from. example format 2020-03-15",
+          "type": "string"
+        },
+        "end": {
+          "description": "The end date to pull data to. example format 2020-03-15",
+          "type": "string"
+        }
+      },
+      "required": [
```

Comment:
consider specifying the regex pattern or a "format" of the property

Reply:
start and end are dates. example validation failed with format as date-time.

Let me know if there is another alternative we can try. @absyed FYI

Reply:
Are they the YYYYMM format dates? You could include a regex to match